internal/v1/handler_compose_image.go

```diff
@@ -726,51 +805,59 @@ func (h *Handlers) buildCustomizations(ctx echo.Context, cr *ComposeRequest, d *
 		res.Packages = cust.Packages
 	}
 
-	snapshotDate := cr.ImageRequests[0].SnapshotDate
-	if cust.PayloadRepositories != nil && snapshotDate != nil {
-		var repoURLs []string
-		var repoIDs []string
-		for _, payloadRepository := range *cust.PayloadRepositories {
-			if payloadRepository.Baseurl != nil {
-				repoURLs = append(repoURLs, *payloadRepository.Baseurl)
-			} else if payloadRepository.Id != nil {
-				repoIDs = append(repoIDs, *payloadRepository.Id)
-			}
-		}
-		payloadRepositories, _, err := h.buildRepositorySnapshots(ctx, repoURLs, repoIDs, true, *snapshotDate)
+	if cust.Template != nil {
+		_, templateRepositories, err := h.buildTemplateRepositories(ctx, *cust.Template)
 		if err != nil {
 			return nil, err
 		}
-		res.PayloadRepositories = &payloadRepositories
-	} else if cust.PayloadRepositories != nil && len(*cust.PayloadRepositories) > 0 {
-		plrepos, err := h.buildPayloadRepositories(ctx, *cust.PayloadRepositories)
-		if err != nil {
-			return nil, err
+		res.CustomRepositories = &templateRepositories
+	} else {
```

Comment:
This isn't correct, there are 3 scenarios:

1. content template specified: use buildTemplateRepositories
2. snapshot date specified: use buildSnapshotRepositories
3. no snapshot date or content template: use the customizations directly

Reply:
hmm in that case, i think maybe i've also missed overwriting the payload repositories (i'm not actually sure what these are :thinking: ) if a template if a specified. i don't think custom repos would be specified if a template is, is that the same for payload repos?